public/app/plugins/datasource/grafana-azure-monitor-datasource/components/VariableEditor/VariableEditor.tsx

```diff
@@ -51,6 +71,17 @@ const VariableEditor = (props: Props) => {
     }
   };
 
+  const onChangeSubscription = (selectableValue: SelectableValue) => {
+    if (selectableValue.value) {
+      const newQuery = {
+        ...query,
+        subscription: selectableValue.value,
+      };
+      setQuery(newQuery);
+      props.onChange(newQuery);
```

Comment:
Would it be a better pattern here to `useEffect(() => props.onChange(newQuery), [newQuery])`?

Reply:
Yup makes sense to me 😊